### PR TITLE
OEL-491:Paragraphs - Listing Item Block - Theming

### PR DESCRIPTION
**Pull request by abel-santos-corral**
Jira issue(s):
- [OEL-491](https://citnet.tech.ec.europa.eu/CITnet/jira/browse/OEL-491)

This branch comes from a POC we did at branch OEL-490.

### CODE REVIEW

File: includes/paragraphs.inc [hunk 1]
```diff
@@ -120,3 +120,131 @@ function oe_bootstrap_theme_preprocess_paragraph__oe_text_feature_media(array &$
...
+
+/**
+ * Implements hook_preprocess_paragraph() for paragraph--oe-list-item-block.html.twig.
+ */
+function oe_bootstrap_theme_preprocess_paragraph__oe_list_item_block(array &$variables): void {
+  /** @var \Drupal\paragraphs\Entity\Paragraph $paragraph */
+  $paragraph = $variables['paragraph'];
+  $variables['title'] = $paragraph->get('field_oe_title')->first()->value;
+  // Set up the correct variant name.
+  $variant_block = $paragraph->get('field_oe_list_item_block_layout')->first()->value;
```
Comment (drishu) [inline includes/paragraphs.inc]: remove this

File: includes/paragraphs.inc [hunk 2]
```diff
@@ -120,3 +120,131 @@ function oe_bootstrap_theme_preprocess_paragraph__oe_text_feature_media(array &$
...
+  set_variant_block($variant_block);
```
Comment (drishu) [inline includes/paragraphs.inc]:
```suggestion
  $variables['columns']= set_variant_block($paragraph->get('field_oe_list_item_block_layout')->first()->value);
```

File: includes/paragraphs.inc [hunk 3]
```diff
@@ -120,3 +120,131 @@ function oe_bootstrap_theme_preprocess_paragraph__oe_text_feature_media(array &$
...
+      'path' => $variables['bcl_icon_path'],
+      'name' => 'chevron-right',
+    ],
+  ];
+}
+
+/**
+ * Sets variant block.
+ */
+function set_variant_block(&$variant_block): void {
```
Comment (drishu) [inline includes/paragraphs.inc]:
```suggestion
function set_variant_block($variant_block): void {
```

File: includes/paragraphs.inc [hunk 4]
```diff
@@ -120,3 +120,131 @@ function oe_bootstrap_theme_preprocess_paragraph__oe_text_feature_media(array &$
...
+    'label' => $link_item->get('title')->getValue(),
+    'icon' => [
+      'path' => $variables['bcl_icon_path'],
+      'name' => 'chevron-right',
+    ],
+  ];
+}
+
+/**
+ * Sets variant block.
```
Comment (drishu) [inline includes/paragraphs.inc]: add parameter and return

File: includes/paragraphs.inc [hunk 5]
```diff
@@ -120,3 +120,131 @@ function oe_bootstrap_theme_preprocess_paragraph__oe_text_feature_media(array &$
...
+}
+
+/**
+ * Sets image position and rounded.
+ */
+function set_image($variant, array &$image): void {
+  if ($variant == 'thumbnail_secondary') {
+    $image['position'] = 'bottom';
+  }
+  elseif ($variant == 'thumbnail_secondary') {
```
Comment (drishu) [inline includes/paragraphs.inc]: its the same condition!

File: includes/paragraphs.inc [hunk 6]
```diff
@@ -120,3 +120,131 @@ function oe_bootstrap_theme_preprocess_paragraph__oe_text_feature_media(array &$
...
+    $image['position'] = 'top';
+  }
+  else {
```
Comment (drishu) [inline includes/paragraphs.inc]: how about a switch case: and default:

File: includes/paragraphs.inc [hunk 7]
```diff
@@ -120,3 +120,131 @@ function oe_bootstrap_theme_preprocess_paragraph__oe_text_feature_media(array &$
...
+  ];
+}
+
+/**
+ * Sets variant block.
+ */
+function set_variant_block(&$variant_block): void {
+  switch ($variant_block) {
+    case 'one_column':
+      $variant_block = '1-col';
```
Comment (drishu) [inline includes/paragraphs.inc]:
```suggestion
      return 'highlight-1-col';
```

File: includes/paragraphs.inc [hunk 8]
```diff
@@ -120,3 +120,131 @@ function oe_bootstrap_theme_preprocess_paragraph__oe_text_feature_media(array &$
...
+  $paragraph = $variables['paragraph'];
+  $variables['title'] = $paragraph->get('field_oe_title')->first()->value;
+  // Set up the correct variant name.
+  $variant_block = $paragraph->get('field_oe_list_item_block_layout')->first()->value;
+  set_variant_block($variant_block);
+  $variables['columns'] = 'highlight-' . $variant_block;
+  $variables['items'] = [];
+
+  foreach ($variables['paragraph']->get('field_oe_paragraphs') as $delta => $item) {
+    $variant = $item->entity->get('oe_paragraphs_variant')->first()->value;
```
Comment (drishu) [inline includes/paragraphs.inc]: remove this line, for now, its always highlighted

File: includes/paragraphs.inc [hunk 9]
```diff
@@ -120,3 +120,131 @@ function oe_bootstrap_theme_preprocess_paragraph__oe_text_feature_media(array &$
...
+    if ($variant == 'highlight') {
+      $text_classes = 'mb-2';
```
Comment (drishu) [inline includes/paragraphs.inc]: only this

File: includes/paragraphs.inc [hunk 10]
```diff
@@ -120,3 +120,131 @@ function oe_bootstrap_theme_preprocess_paragraph__oe_text_feature_media(array &$
...
+ */
+function set_image($variant, array &$image): void {
+  if ($variant == 'thumbnail_secondary') {
+    $image['position'] = 'bottom';
+  }
+  elseif ($variant == 'thumbnail_secondary') {
+    $image['position'] = 'top';
+  }
+  else {
+    $image['rounded'] = 1;
```
Comment (drishu) [inline includes/paragraphs.inc]: top or bottom or rounded ? we need a default one, top, then if variant is secondary show bottom

**Review by drishu — CHANGES_REQUESTED** [review]

**Review by drishu — APPROVED** [review]

**Review by escuriola — APPROVED** [review]

**Pull request merged**